kratos/includes/exception.h

```diff
@@ -24,6 +24,7 @@
 // Project includes
 #include "includes/kratos_export_api.h"
 #include "includes/code_location.h"
+#include "utilities/stl_io.h" // Defines operator<< for several stl types, needed to print such types in error messages using Exception::operator<<
```

Comment:
Doing this will basically include many stl-containers everywhere (through „define.h“)

Reply:
Note that you are already including <vector> and <iostream> in the include block just above (so we are adding <set> and <memory> to the list of headers)

Reply:
Just to say that I agree with @jcotela . There is not so much difference and right now the compilation with Clang is broken. I'd merge this as it is. We can always find a more efficient solution afterwards.

Reply:
Have you tried including the stl io in the test where the compilation fails?

(Brief answer, don’t have much time)
To be discussed next week

Reply:
The compilation fails in a test... This would imply to include it in every place we use the macro. Am I wrong?

Reply:
Yes exactly

Reply:
Taking into account that It only crashes with Clang, I found this not so robust...

Reply:
> Have you tried including the stl io in the test where the compilation fails?

Yes. Getting the `operator<<` of `Exception` to accept stl containers is not a problem, it does already. The problem is that the generic implementation of `Exception::operator<<` contains a call to to `operator<<` for streams and the argument type, which requires the definitions in `stl_io` for stl containers.

>  The compilation fails in a test... This would imply to include it in every place we use the macro. Am I wrong?

Only if using std::vector (the macro should work fine for array_1d, ublas types, etc.)